src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ShopService;
+import java.util.Map;
+import java.util.Set;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport(ShopService service) {
+        Storage storage = service.getStorage();
```

Comment:
Directly accessing the storage from the ReportGenerator violates the SOLID principles, specifically the Dependency Inversion Principle. Instead, the ShopService should provide the necessary data to generate the report, without exposing the storage implementation.